go/channelling/roomworker.go

```diff
@@ -80,49 +77,33 @@ func NewRoomWorker(manager *roomManager, roomID, roomName, roomType string, cred
 		name:     roomName,
 		roomType: roomType,
 		workers:  make(chan func(), roomMaxWorkers),
-		expired:  make(chan bool),
+		expired:  make(chan bool, 1),
 		users:    make(map[string]*roomUser),
 	}
 
 	if credentials != nil && len(credentials.PIN) > 0 {
 		r.credentials = credentials
 	}
 
-	// Create expire timer.
-	r.timer = time.AfterFunc(roomExpiryDuration, func() {
-		r.expired <- true
-	})
-
 	return r
 }
 
 func (r *roomWorker) Start() {
 	// Main blocking worker.
 L:
 	for {
-		r.timer.Reset(roomExpiryDuration)
 		select {
 		case w := <-r.workers:
 			//fmt.Println("Running worker", r.Id, w)
+			// TODO(leon): All workers run in this main blocking worker thread. Good?
```

Comment:
Didn't understand the comment, but the code. It still isn't clear to me why the channel is named `workers`, should have been named `funcQueue` or similar to indicate that it only stores / transfers functions to be executed by one _single_ thread.